gobblin-compaction/src/main/java/gobblin/compaction/event/CompactionSlaEventHelper.java

```diff
@@ -56,8 +59,19 @@ public static void setUpstreamTimeStamp(State state, long time) {
   }
 
   private static void setDatasetUrn(Dataset dataset) {
-    dataset.jobProps().setProp(SlaEventKeys.DATASET_URN_KEY,
-        new Path(dataset.jobProps().getProp(MRCompactor.COMPACTION_DEST_DIR), dataset.topic()).toString());
+    dataset.jobProps().setProp(SlaEventKeys.DATASET_URN_KEY, getDatasetUrnFromDataset(dataset));
+  }
+
+  private static String getDatasetUrnFromDataset(Dataset dataset) {
```

Comment:
This implementation is too specific. You can move this method to `Dataset`, so if someone wants to define dataset urn differently, he can override it in a subclass of `Dataset`.